feat: introduce learn v2 behind a feature-flag

GuiBibeau opened the pull request:
### Problem

The current learn section is dull and offers not many informations to learners. This introduces behind a feature flag a new learn v2 section.

### Summary of Changes

- introduce a feature flag that when turned off redirects to `/learn` allowing us to work in multiple incremental PRs
- proposes a layout for the learn section.
- The styling is not final as this PR is meant to be a first stab at the section. 

https://github.com/user-attachments/assets/56c5896f-9d56-4a59-821e-f2e9cd83af9b





Fixes #

h4rkl commented on src/app/[locale]/learn-v2/page.tsx:
Use `/learn` for route and update components to use existing component libs wherever possible rather than custom code to benefit from future upgrade paths.

jacobcreech commented on assets/developers/hero-solana-learn.png:
This likely wouldn't pass design review. Logo is quite a bit off and Raj will come after you.

GuiBibeau commented on src/components/header/HeaderListCommunity.js:
```diff
@@ -2,6 +2,7 @@ import { useTranslation } from "next-i18next";
 import Link from "../../utils/Link";
 import BreakpointLogo from "../../../assets/nav/community/breakpoint-logo.inline.svg";
 import InvolvedSVG from "../../../assets/nav/community/involved.inline.svg";
+import Dropdown from "react-bootstrap/Dropdown";
```
The only change i introduced was new links for the learn dropdown and fixing a bug where the dropdown was not closing on an internal redirect. I don't mind taking on the migration of the menus but it significantly introduces scope to an already big PR. 

What's your take on the best course of action?

GuiBibeau commented:
Changes since last reviews:

- new hero
- accessibility checks
- new cards
- simpler layout
- removed the drop down menu,
- now the top nav is really just a link to the learn section
- added a new full post  explaining what is solana
- No wording of web3, blockchain. Just Solana on it's own as a concept.
- less finance, more approachable and varied
outcome focused

h4rkl commented on src/components/header/HeaderList.js:
```diff
@@ -45,29 +44,15 @@ const HeaderList = () => {
 
   return (
     <ul className="navbar-nav ms-auto">
-      <Dropdown
```
Keep the dropdown, add 1st link as "Learn Home", and then list the 10 links in the dropdown. Master plan is to rework the whole header but for now lets keep it consistent. Important for learn items to be directly accessible in the header menu from other pages.

h4rkl replied:
You'll likely see less SEO funnels into the Learn content. Also surfacing links in the header menu gives them more weight. But this isn't a show stopper.

GuiBibeau replied:
I'll do a dedicated follow up PR on this to come up with a compromise. Maybe select 3-4 most important posts and have the learning center as a main link.

h4rkl commented on src/app/[locale]/learn/learn-page-content.tsx:
```diff
@@ -0,0 +1,228 @@
+            </p>
+            <ul className="space-y-3" role="list">
+              <li>
+                <Link
```
Make at least the first span in these links a different color treatment from the text to meet WCAG guidelines

h4rkl commented on src/app/[locale]/learn/learn-page-content.tsx:
```diff
@@ -0,0 +1,228 @@
+"use client";
```
make a `/learn` folder in components and move learn-page-content.tsx

h4rkl commented on src/app/[locale]/learn/learn-page-content.tsx:
```diff
@@ -0,0 +1,228 @@
+            <p className="text-gray-300 mb-8">
+              {translations.continueJourneySubtitle}
+            </p>
+            <ul className="space-y-3" role="list">
```
This whole section needs design treatment

<img width="1358" alt="Screenshot 2025-07-09 at 8 31 19 PM" src="https://github.com/user-attachments/assets/091fa3ab-863e-4fc3-9227-261043d3b548" />

GuiBibeau commented:
After a bit of back and forth on design, we are going back to reusing some components from /developers

ZYJLiu commented on content/learn/en/what-are-nfts.mdx:
```diff
@@ -0,0 +1,202 @@
+---
+title: What are NFTs?
```
Maybe limit the content of this page to explaining NFTs are unique tokens and how to create them, and remove sections related to how to buy NFTs and "evaluating" NFT projects.

Even though you've mentioned "Red Flags to Avoid", I think it would be better avoid mentioning buying NFTs on solana-com content.

ZYJLiu commented on content/learn/en/introduction-to-defi-on-solana.mdx:
```diff
@@ -0,0 +1,198 @@
+- Factor in token price changes
+- Compare to simpler alternatives
+
+## Common DeFi Strategies
```
Maybe remove this section on "DeFi Strategies". 

I don't particularly like the idea of suggesting for users to go try Defi protocols on solana-com content. "Defi" is such a broad concept that and the risks (as mentioned on this page) users take on for any given protocol yield is not well defined and difficult to evaluate.

GuiBibeau replied:
That's a great point! I'll remove that

ZYJLiu commented:
Just read through parts 1-10 and its a great outline of the basics!
The default font size feels a bit small, but the overall UI looks good. Maybe add a regular table of contents that shows the h2 content of each page and convert the "Chapters" nav on the right hand side to a dropdown.
![CleanShot 2025-07-09 at 22 33 45@2x](https://github.com/user-attachments/assets/8b328f67-b911-4285-bef6-cdb73df6a141)
